api/messages.go

```diff
@@ -28,14 +29,41 @@ func (br *decodingError) Error() string {
 }
 
 type CreateCredentialRequest struct {
-	Address string `json:"address"`
-	Msg     string `json:"msg"`
-	Sig     string `json:"sig"`
-	Version string `json:"version"`
+	Address common.Address `json:"address"`
+	Msg     []byte         `json:"msg"`
+	Sig     []byte         `json:"sig"`
+	Version string         `json:"version"`
 
 	operatorType credentials.OperatorType `json:"-"`
 }
 
+func (c *CreateCredentialRequest) UnmarshalJSON(data []byte) error {
+	type Alias CreateCredentialRequest
+	aux := &struct {
+		Address string `json:"address"`
+		Msg     string `json:"msg"`
+		Sig     string `json:"sig"`
+		*Alias
```

Comment:
this field is actually unused and can be removed

Reply:
oh nvm i guess putting it here populates `Version`

Reply:
yep